incubator/group/handler.go

```diff
@@ -0,0 +1,50 @@
+package group
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+)
+
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case MsgCreateGroup:
+			return handleMsgCreateGroup(ctx, k, msg)
+		//case MsgUpdateGroupMembers:
+		//	return handleMsgUpdateGroupMembers(ctx, k, msg)
+		//case MsgUpdateGroupAdmin:
+		//	return handleMsgUpdateGroupAdmin(ctx, k, msg)
+		//case MsgUpdateGroupComment:
+		//	return handleMsgUpdateGroupComment(ctx, k, msg)
+		default:
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized group message type: %T", msg)
+		}
+	}
+
+}
+
+func handleMsgCreateGroup(ctx sdk.Context, k Keeper, msg MsgCreateGroup) (*sdk.Result, error) {
+	id, err := k.CreateGroup(ctx, msg.Admin, msg.Members, msg.Comment)
+	if err != nil {
+		return nil, errors.Wrap(err, "create group")
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, ModuleName),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Admin.String()),
+		),
+	)
+
+	//TODO: the data/ log response is not specified
```

Comment:
🤔 If we would use protobuf for the GroupID then clients would have to encode the value for queries as well. The stored GroupID in the table and the entity would not match the encoded value unless we also switch to storing protobuf there, too. To be honest, I would prefer to always return the raw key that matches the table key so that clients can copy it for queries without thinking about encoding.

Reply:
This is maybe a larger discussion about how to enable easy client queries with protobuf. Either way we should be obvious for clients both what the return value is and how to do store queries. Both of which could use a bit more supporting infrastructure

Reply:
discussed -> raw key